stripe/api_resources/abstract/deletable_api_resource.py

```diff
@@ -3,16 +3,23 @@
 from stripe import util
 from stripe.api_resources.abstract.api_resource import APIResource
 from urllib.parse import quote_plus
+from typing import TypeVar, cast
+from stripe.stripe_object import StripeObject
 
+T = TypeVar("T", bound="StripeObject")
 
-class DeletableAPIResource(APIResource):
+
+class DeletableAPIResource(APIResource[T]):
     @classmethod
     def _cls_delete(cls, sid, **params):
```

Comment:
Type this one as well?

Reply:
I'm not sure if it helps with all the `class_method_variant` magic.

Reply:
Can't hurt.
```suggestion
    def _cls_delete(cls, sid, **params) -> T:
```